Lib/unittest/mock.py

```diff
@@ -2035,9 +2035,9 @@ def __new__(cls, value=(), name='', parent=None, two=False,
 
     def __init__(self, value=(), name=None, parent=None, two=False,
                  from_kall=True):
-        self.name = name
-        self.parent = parent
-        self.from_kall = from_kall
+        self._mock_name = name
+        self._mock_parent = parent
+        self._mock_from_kall = from_kall
```

Comment:
This is the fix I had in mind, but it needs unit tests to show why it's needed.

Reply:
Done.